packages/amplify-category-function/src/provider-utils/awscloudformation/utils/environmentVariablesHelper.ts

```diff
@@ -0,0 +1,116 @@
+import path from 'path';
+import _ from 'lodash';
+import { $TSContext, pathManager, JSONUtilities } from 'amplify-cli-core';
+import { categoryName, functionParametersFileName } from './constants';
+
+export const getStoredEnvironmentVariables = (context: $TSContext, resourceName: string): object => {
+  const storedList = getStoredList(resourceName);
+  const storedReferences = getStoredReferences(resourceName);
+  const storedKeyValue = getStoredKeyValue(context, resourceName);
+  const environmentVariables = {};
+
+  // Search for key and take matched
+  // environment variables based on 4 steps
+  storedList.forEach(({ environmaneVariableName, cloudFormationParameterName }) => {
+    // 1. Check if format is correct in function-parmeter.json
+    if (!environmaneVariableName) return;
+    if (!cloudFormationParameterName) return;
+    // 2. Check if key exists in CFn template
+    if (!storedReferences[environmaneVariableName]) return;
+    if (!storedReferences[environmaneVariableName].Ref) return;
+    if (storedReferences[environmaneVariableName].Ref !== cloudFormationParameterName) return;
+    // 3. Check if key exists in team-provider-info.json
+    if (!_.has(storedKeyValue, cloudFormationParameterName)) return;
+
+    // 4. Take matched environment variables
+    environmentVariables[environmaneVariableName] = storedKeyValue[cloudFormationParameterName];
+  });
+
+  return environmentVariables;
+};
+
+export const setEnvironmentVariable = (context: $TSContext, resourceName: string, newEnvironmentVariable: object): void => {
+  const newList = getStoredList(resourceName);
+  const newReferences = getStoredReferences(resourceName);
+  const newKeyValue = getStoredKeyValue(context, resourceName);
+  _.each(newEnvironmentVariable, (value, key) => {
+    const cameledKey = _.camelCase(key);
+    newList.push({
+      cloudFormationParameterName: cameledKey,
+      environmaneVariableName: key,
```

Comment:
Should this be `environmentVariableName` throughout the PR?

Reply:
Right, I updated the key name in all places. Thank you :)